### PR TITLE
PYTHON-4673 - Add Async Encryption Tests

### PR DESCRIPTION
**Pull request by NoahStapp**

### CODE REVIEW

File: test/asynchronous/test_encryption.py [hunk 1]
```diff
@@ -0,0 +1,3155 @@
...
+
+AWS_TEMP_NO_SESSION_CREDS = {
+    "accessKeyId": os.environ.get("CSFLE_AWS_TEMP_ACCESS_KEY_ID", ""),
+    "secretAccessKey": os.environ.get("CSFLE_AWS_TEMP_SECRET_ACCESS_KEY", ""),
+}
+KMS_TLS_OPTS = {"kmip": {"tlsCAFile": CA_PEM, "tlsCertificateKeyFile": CLIENT_PEM}}
+
+
+if _IS_SYNC:
+    # TODO: Add asynchronous SpecRunner
```
[the remark this answers is not in the document]
Comment (NoahStapp) [reply test/asynchronous/test_encryption.py:615]: Yes, that would be a significantly larger change.

[the remark this answers is not in the document]
Comment (blink1073) [reply test/asynchronous/test_encryption.py:615]: Can you create it and put a reference here?

[the remark this answers is not in the document]
Comment (NoahStapp) [reply test/asynchronous/test_encryption.py]: Done.

**Review by blink1073 — APPROVED** [review]
LGTM!

**Pull request merged**